component/organism/AgencyForm/AgencyForm.js

```diff
@@ -0,0 +1,134 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Button, Input } from "../../atoms";
+import { useForm } from "react-hook-form";
+import styles from "./AgencyForm.module.css";
+
+const AgencyForm = (props) => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({ reValidateMode: "onSubmit" });
+
+  function submitRequest(e) {
+    // e.preventDefault();
+    console.log(e);
+  }
+  console.log(errors);
+
+  return (
+    <div className={styles.containerMaster}>
+      <form
+        className={styles.container}
+        onSubmit={handleSubmit(async (data, e) => {
+          const reqType = e.nativeEvent.submitter.innerHTML;
+          console.log("data", data);
+          const res = await fetch("api/agencies/agencies", {
+            method: reqType,
+            body: JSON.stringify(data),
+          });
+          const json = await res;
+          console.log("json", json);
+        })}
+        id="crud"
+      >
+        <div className={styles.formContainer}>
+          <Input
+            type="text"
+            register={{
+              ...register("name", { required: "This is field is required" }),
+            }}
+            label="Company name"
+          />
+          {errors.name?.message}
+          <Input
+            type="text"
+            register={{
+              ...register("city", {
+                required: "This is field is required",
+              }),
+            }}
+            label="City"
+          />
+          {errors.city?.message}
+          <Input
+            type="text"
+            register={{
+              ...register("region", { required: "This is field is required" }),
+            }}
+            label="Region"
+          />
+          {errors.region?.message}
+          <Input
+            type="text"
+            register={{
+              ...register("size", { required: "This is field is required" }),
+            }}
+            label="Company size"
+          />
+          {errors.size?.message}
+          <Input
+            type="text"
+            register={{
+              ...register("website", { required: "This is field is required" }),
```

Comment:
We should probably validate that this input is a url
I suggesting installing Yup, creating a schema and to let yup create the error messages for you